pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/LoadManagerShared.java

```diff
@@ -409,22 +413,23 @@ private static void filterDomainsNotHavingLeastNumberAntiAffinityNamespaces(
             return nsCount != null && nsCount != finalLeastNamespaceCount;
         });
     }
-
+    @FunctionalInterface
+    private interface CountAntiAffinityNamespaceOwnedBrokers{
+        void run(String broker, String namespace);
+    }
     /**
      * It returns map of broker and count of namespace that are belong to the same anti-affinity group as given.
      *
      * @param pulsar
      * @param namespaceName
-     * @param brokerToNamespaceToBundleRange
+     * @param bundleOwnershipData
      * @return
      */
     public static CompletableFuture<Map<String, Integer>> getAntiAffinityNamespaceOwnedBrokers(
             final PulsarService pulsar, final String namespaceName,
-            final ConcurrentOpenHashMap<String, ConcurrentOpenHashMap<String, ConcurrentOpenHashSet<String>>>
-                    brokerToNamespaceToBundleRange) {
+            Object bundleOwnershipData) {
```

Comment:
It's confusing to add an Object type param. Actually, the method needs a list of data pairs (broker -> namespace), maybe the caller can filter invalid data, and prepare the data first.

For ModularLoadManagerImpl, convert ConcurrentOpenHashMap to a List, and filter empty bundle ranges, then call this method.
For ExtensibleLoadManagerImpl, convert Set to a List, filter bundles with an invalid state, then call this method.

Reply:
+1

Reply:
This ownershipData can be large. It could unnecessarily increase space complexity if we convert them to the temporary list just to reuse this function. We can have overloaded functions(with minimal duplicated code) for these different ownership data structures.

I added the overloaded functions.

Reply:
OK, thanks.